packages/block-editor/src/components/block-canvas/index.js

```diff
@@ -32,10 +35,22 @@ export function ExperimentalBlockCanvas( {
 	iframeProps,
 } ) {
 	useBlockCommands();
+	const isTabletViewport = useViewportMatch( 'medium', '<' );
 	const resetTypingRef = useMouseMoveTypingReset();
 	const clearerRef = useBlockSelectionClearer();
 	const localRef = useRef();
 	const contentRef = useMergeRefs( [ contentRefProp, clearerRef, localRef ] );
+	const isZoomedOut = useSelect(
+		( select ) => unlock( select( blockEditorStore ) ).isZoomOut(),
+		[]
+	);
+	const zoomOutIframeProps =
+		isZoomedOut && ! isTabletViewport
+			? {
+					scale: 'default',
+					frameSize: '40px',
+			  }
+			: {};
```

Comment:
This is the main change in this PR, basically the zoom-out behavior is absorbed into the block-editor.